redis.go

```diff
@@ -0,0 +1,230 @@
+package mercure
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/redis/go-redis/v9"
+	"golang.org/x/net/context"
+)
+
+const (
+	lastEventIDKey = "lastEventID"
+	publishScript  = `
+		redis.call("SET", KEYS[1], ARGV[1])
+		redis.call("PUBLISH", "*", ARGV[2])
+		return true
+	`
+)
+
+type RedisTransport struct {
+	sync.RWMutex
+	logger                       Logger
+	client                       *redis.Client
+	subscribers                  *SubscriberList
+	subscribersBroadcastParallel int
+	dispatcher                   chan SubscriberPayload
+	closed                       chan any
+	publishScript                *redis.Script
+}
+
+type SubscriberPayload struct {
+	subscriber *LocalSubscriber
+	payload    Update
+}
+
+func NewRedisTransport(logger Logger, address string, username string, password string, subscribersSize int, subscribersBroadcastParallel int) (*RedisTransport, error) {
+	client := redis.NewClient(&redis.Options{
+		Username: username,
+		Password: password,
+		Addr:     address,
+	})
+
+	return NewRedisTransportInstance(logger, client, subscribersSize, subscribersBroadcastParallel)
+}
+
+func NewRedisTransportInstance(
+	logger Logger,
+	client *redis.Client,
+	subscribersSize int,
+	subscribersBroadcastParallel int,
+) (*RedisTransport, error) {
+	subscriber := client.PSubscribe(context.Background(), "*")
```

Comment:
would it make sense to check that we have a connection?
```
pong, err := rdb.Ping(ctx).Result()
```

Also, do we need to use wildcard here? Maybe something more specific or even make it configurable?